### PR TITLE
Update AppInsights scheduler tracking

### PR DESCRIPTION
**Pull request by doncem**
### Change description ###

Majorly drugging up task scheduler to track each schedule automatically. This eventually introduces modifications to "serivce up" alert and potentially replaces unnecessary log messages of particular task being run

This is official [Microsoft](github.com/Microsoft/ApplicationInsights-Java) suggestion

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by doncem** [comment]
For a bit enabled scheduling in demo and ran this query

```dql
requests
| where name startswith "Sch"
| project timestamp, name, success, duration
```

top 25 results:

```csv
timestamp,name,success,duration
"2019-05-07T10:14:15.94Z","Schedule /UploadLettersTask",True,0
"2019-05-07T10:13:45.926Z","Schedule /UploadLettersTask",True,0
"2019-05-07T10:13:15.903Z","Schedule /UploadLettersTask",True,0
"2019-05-07T10:12:15.864Z","Schedule /UploadLettersTask",True,0
"2019-05-07T10:10:45.8Z","Schedule /UploadLettersTask",True,0
"2019-05-07T10:10:15.779Z","Schedule /UploadLettersTask",True,0
"2019-05-07T10:09:45.767Z","Schedule /UploadLettersTask",True,0
"2019-05-07T10:09:15.737Z","Schedule /UploadLettersTask",True,9
"2019-05-07T10:06:15.634Z","Schedule /UploadLettersTask",True,0
"2019-05-07T10:05:45.612Z","Schedule /UploadLettersTask",True,16
"2019-05-07T10:05:15.586Z","Schedule /UploadLettersTask",True,0
"2019-05-07T10:04:45.551Z","Schedule /UploadLettersTask",True,16
"2019-05-07T10:04:15.535Z","Schedule /UploadLettersTask",True,0
"2019-05-07T10:03:45.504Z","Schedule /UploadLettersTask",True,16
"2019-05-07T10:03:15.487Z","Schedule /UploadLettersTask",True,0
"2019-05-07T10:02:45.462Z","Schedule /UploadLettersTask",True,0
"2019-05-07T10:02:15.44Z","Schedule /UploadLettersTask",True,0
"2019-05-07T10:01:45.415Z","Schedule /UploadLettersTask",True,0
"2019-05-07T10:01:15.399Z","Schedule /UploadLettersTask",True,0
"2019-05-07T10:00:45.378Z","Schedule /UploadLettersTask",True,0
"2019-05-07T10:00:15.343Z","Schedule /UploadLettersTask",True,16
"2019-05-07T10:00:00.013Z","Schedule /MarkLettersPostedTask",True,8813
"2019-05-07T09:59:45.328Z","Schedule /UploadLettersTask",True,0
"2019-05-07T09:59:15.312Z","Schedule /UploadLettersTask",True,0
"2019-05-07T09:58:45.299Z","Schedule /UploadLettersTask",True,0
```

File: src/main/java/uk/gov/hmcts/reform/sendletter/config/AspectConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            handleRequestTelemetry(requestTelemetry, joinPoint.getTarget().getClass().getSimpleName(), success);
+        }
+    }
+
+    private void handleRequestTelemetry(
+        RequestTelemetryContext requestTelemetryContext,
+        String caller,
+        boolean success
+    ) {
+        if (requestTelemetryContext != null) {
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/reform/sendletter/config/AspectConfiguration.java]: when can this requestTelemetryContext be null ?

Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/sendletter/config/AspectConfiguration.java]: `ThreadContext` has possibility of removing telemetry context hence possibility of it being null. Just to be safe from that - there's a wrapper

Comment (nitinprabhu) [reply src/main/java/uk/gov/hmcts/reform/sendletter/config/AspectConfiguration.java]: ok can we at least log the fact that it is null ?

File: src/main/java/uk/gov/hmcts/reform/sendletter/config/ThreadPoolConfig.java [hunk 2]
```diff
@@ -1,35 +1,126 @@
...
         });
+        taskScheduler.initialize();
+
+        taskRegistrar.setTaskScheduler(taskScheduler);
+    }
+
+    /**
+     * Custom {@link ThreadPoolTaskScheduler} to be able to register scheduled tasks via AppInsights.
+     */
+    private static class SendLetterTaskScheduler extends ThreadPoolTaskScheduler {
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/reform/sendletter/config/ThreadPoolConfig.java]: Worth putting a link in the comment https://github.com/Microsoft/ApplicationInsights-Java/wiki/Distributed-Tracing-in-Asynchronous-Java-Applications

**Pull request merged**